### PR TITLE
Add module makefile

### PR DESCRIPTION
**Pull request by TomDonoghue**
Add's a makefile, to work with the module, that is designed to replace `summary.sh`. 

Goals for review:
- can you run all the make targets, and do they seem to work?
- any thought on naming structure, organizations, additions, or anything else for this?

Note - this idea partially based on this example:
https://github.com/nltk/nltk/blob/develop/Makefile

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,97 @@
+# Makefile for working with a Python module
+
+##########################################################################
+## REQUIREMENTS
+#
+# The following packages are required to run this makefile:
+#   pytest 				For running tests
+#   coverage 			For running test coverage
+#   cloc 				For counting code
```
Comment (ryanhammonds) [inline Makefile:9]: Should shell/bash installation be specified? `cloc` can be installed via pip, but it required a system/os package installation for me.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+# Run all counts
+run_counts:
+	@make count_size
+	@make count_cloc
+	@make count_cloc_tests
+
+# Count the total number of lines in the module
+count_size:
+	@echo "\n\n CHECK MODULE SIZE:"
```
[the remark this answers is not in the document]
Comment (TomDonoghue) [reply Makefile:31]: Ahh, good catch! I think it ended up echo from tweaking some test cases - I switched back to printf

[the remark this answers is not in the document]
Comment (TomDonoghue) [reply Makefile:31]: As to why - as I understand it `printf` is `print-formatted`, and so interprets the `\n`, where as I think echo is just `print literal`, so doesn't do any formatting.

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,97 @@
+# Makefile for working with a Python module
+
+##########################################################################
+## REQUIREMENTS
+#
+# The following packages are required to run this makefile:
+#   pytest 				For running tests
+#   coverage 			For running test coverage
```
[the remark this answers is not in the document]
Comment (TomDonoghue) [reply Makefile:8]: Ooops - yeh. I learnt in this process that `make` is _very_ picky about tabs, but tab spacing went weird here. Fixed.

File: Makefile (hunk 1 above)
Comment (TomDonoghue) [reply Makefile:9]: Ahh, yes, very good point -  it's worth pointing out that most of these are Python packages, where as cloc is a command line tool (it's not the same tool as the `cloc` that is on PYPI I think). I've split out the requirements, to point out what kind of thing cloc is.

**Comment by TomDonoghue** [comment]
Thanks for the review @ryanhammonds! Updated for comments, so I'm going to merge!

**Pull request merged**